clients/src/test/java/org/apache/kafka/common/record/MemoryRecordsBuilderTest.java

```diff
@@ -504,20 +510,21 @@ public void writePastLimit() {
         }
     }
 
-    @Test(expected = IllegalArgumentException.class)
+    @Test
     public void testAppendAtInvalidOffset() {
         ByteBuffer buffer = ByteBuffer.allocate(1024);
         buffer.position(bufferOffset);
 
         long logAppendTime = System.currentTimeMillis();
-        MemoryRecordsBuilder builder = new MemoryRecordsBuilder(buffer, RecordBatch.MAGIC_VALUE_V1, compressionType,
+        MemoryRecordsBuilder builder = new MemoryRecordsBuilder(buffer, RecordBatch.MAGIC_VALUE_V2, compressionType,
```

Comment:
Why did we change the magic value version here?

Reply:
magic 1 does not support ```ZStandard```

Reply:
Unrelated to the changes in this PR then, just something you noticed? Worth a line in the PR description to avoid confusion.

Reply:
will copy that